extension/basicauth/basicauth_test.go

```diff
@@ -0,0 +1,129 @@
+		{"a", "$2a$10$k87L/MF28Q673VKh8/cPi.SUl7MU/rWuSiIDDFayrKk/1tBsSQu4u"},
+		{"a", "$2a$12$8NJH3LsPrANStV6XtBakCez0cKHXVxmvxIlcz785vxAIZrihHZpeS"},
+		{"abc", "$2a$06$If6bvum7DFjUnE9p2uDeDu0YHzrHM6tf.iqN8.yx.jNN1ILEf7h0i"},
+		{"abcdefghijklmnopqrstuvwxyz", "$2a$06$.rCVZVOThsIa97pEDOxvGuRRgzG64bvtJ0938xuqzv18d3ZpQhstC"},
+		{"abcdefghijklmnopqrstuvwxyz", "$2a$08$aTsUwsyowQuzRrDqFflhgekJ8d9/7Z3GV3UcgvzQW3J5zMyrTvlz."},
+		{"abcdefghijklmnopqrstuvwxyz", "$2a$10$fVH8e28OQRj9tqiDXs1e1uxpsjN0c7II7YPKXua2NAKYvM6iQk7dq"},
+		{"abcdefghijklmnopqrstuvwxyz", "$2a$12$D4G5f18o7aMMfwasBL7GpuQWuP3pkrZrOAnqP.bmezbMng.QwJ/pG"},
+		{"~!@#$%^&*()      ~!@#$%^&*()PNBFRD", "$2a$06$fPIsBO8qRqkjj273rfaOI.HtSV9jLDpTbZn782DC6/t7qT67P6FfO"},
+		{"~!@#$%^&*()      ~!@#$%^&*()PNBFRD", "$2a$08$Eq2r4G/76Wv39MzSX262huzPz612MZiYHVUJe/OcOql2jo4.9UxTW"},
+		{"~!@#$%^&*()      ~!@#$%^&*()PNBFRD", "$2a$10$LgfYWkbzEvQ4JakH7rOvHe0y8pHKF9OaFgwUZ2q7W2FFZmZzJYlfS"},
+		{"~!@#$%^&*()      ~!@#$%^&*()PNBFRD", "$2a$12$WApznUOJfkEGSmYRfnkrPOr466oFDCaj4b6HY3EXGvfxm43seyhgC"},
+		{"ππππππππ", "$2a$10$.TtQJ4Jr6isd4Hp.mVfZeuh6Gws4rOQ/vdBczhDx.19NFK0Y84Dle"},
+	}
+)
+
+func TestBasicAuth_Valid(t *testing.T) {
+	t.Parallel()
+	f, err := ioutil.TempFile("", "htpasswd")
+	require.NoError(t, err)
+	defer os.Remove(f.Name())
+
+	for _, c := range credentials {
+		_, err = fmt.Fprintf(f, "%s:%s\n", c[0], c[1])
+		require.NoError(t, err)
+	}
+
+	ctx := context.Background()
+
+	ba := BasicAuth{
+		htpasswd: HtpasswdSettings{
+			File: f.Name(),
+		},
+	}
+	require.NoError(t, ba.Start(ctx, componenttest.NewNopHost()))
+
+	for _, c := range credentials {
+		t.Run(c[0], func(t *testing.T) {
+			t.Parallel()
+			auth := fmt.Sprintf("%s:%s", c[0], c[0])
+			auth = base64.StdEncoding.EncodeToString([]byte(auth))
+
+			authCtx, err := ba.Authenticate(ctx, map[string][]string{"authorization": {"Basic " + auth}})
+			assert.NoError(t, err)
+			cl := client.FromContext(authCtx)
+			assert.Equal(t, c[0], cl.Auth.GetAttribute("subject"))
+			assert.Equal(t, auth, cl.Auth.GetAttribute("raw"))
+		})
+	}
+}
+
+func TestBasicAuth_InvalidCredentials(t *testing.T) {
+	ba := BasicAuth{
+		htpasswd: HtpasswdSettings{
+			Inline: "username:password",
+		},
+	}
+	require.NoError(t, ba.Start(context.Background(), componenttest.NewNopHost()))
+	_, err := ba.Authenticate(context.Background(), map[string][]string{"authorization": {"Basic dXNlcm5hbWU6cGFzc3dvcmR4eHg="}})
+	assert.Equal(t, errInvalidCredentials, err)
+}
+
+func TestBasicAuth_NoHeader(t *testing.T) {
+	var ba BasicAuth
+	_, err := ba.Authenticate(context.Background(), map[string][]string{})
+	assert.Equal(t, errNoAuth, err)
+}
+
+func TestBasicAuth_InvalidPrefix(t *testing.T) {
+	var ba BasicAuth
+	_, err := ba.Authenticate(context.Background(), map[string][]string{"authorization": {"Bearer token"}})
+	assert.Equal(t, errInvalidSchemePrefix, err)
+}
+
+func TestBasicAuth_InvalidFormat(t *testing.T) {
```

Comment:
Can you please add a test that verifies that the inline takes precedence over the file when the same username is present in both?

Reply:
Done.